### PR TITLE
Ui tests

### PR DESCRIPTION
**Pull request by tolgaarikan**
Hey team, here is the chapter for UI tests. Please take a look and review when you have time. 

Thanks! ☺️

### CODE REVIEW

File: 8-ui-tests.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+UI tests tools that we get with Xcode rests upon two core technologies: the XCTest framework and Accessibility.
+
+- __XCTest__ provides the framework for UI testings, expanding on the same tools you already know from your Unit tests, like XCTAssert and all its relatives!
+
+- __Accessibility__. Yes, you might be asking yourself "why this guy started talking about Accessibility" right now, but it is not only related, but crucial to have UI tests! We use Accessibility tools that are normally are used for providing disabled users a good experience on iOS to also detect what is exposed for external use (the views that on the screen, interactable or not) and use that information for testing.
+
+Main reason we need to use Accessibility to extract information about the UI state of the app is that UI testing is a [black-box testing](https://en.wikipedia.org/wiki/Black-box_testing) framework. We shouldn't have to know anything about the implementation of the code we are testing. We can think of UI testing from the perspective of the user. The user doesn't care how our `MassiveViewController` works (or even that it exists 😈), so why should the UI Tests?
```
Comment (arthur-here) [inline 8-ui-tests.md]: ahaha 😂

File: 8-ui-tests.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+## Getting Started
+
+If your project doesn't already have a target for UI tests, you can add one by going to `File > New > Target..` in Xcode and select a “UI testing bundle”. Then edit your app’s scheme to run your UI tests when testing, by going to `Product > Scheme > Edit Scheme..` in Xcode and adding your UI testing bundle under “Test”.
+
+Before we start with and example, let's talk about using Accessibility to identify views in UI tests. To give our UI elements accesibility identifiers, we can either use interface builder or do it in the implementation code.
+  - To use interface builder to give an identifier we need to select the view, go to `Identity inspector` tab and set a value for `Identifier` field located in the `Accessibility` section.
+  - To give the identifier in code, set `accessibilityIdentifier` of the view to the identifier string you want like following: `anyView.accessibilityIdentifier = "someIdentifierString"`
```
[the remark this answers is not in the document]
Comment (tolgaarikan) [reply 8-ui-tests.md]: Yeah totally. Thanks, will add! :)

File: 8-ui-tests.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
+# UI Tests 📱
+
+This chapter of the cookbook can be considered as `exploration stage` for the following reasons:
+
+- First is that the author does not have a lot of experience with UI tests, 
+- Second, this chapter only talks about the UI test tools that comes bundled with Xcode, and not some of the other options like [iOS Snapshot Test Case](https://github.com/uber/ios-snapshot-test-case), [EarlGrey](https://github.com/google/EarlGrey) and [KIF](https://github.com/kif-framework/KIF),
+- And the last is that the chapter ends with an opinionated paragraph, and might not show the opinion of the whole team.
```
[the remark this answers is not in the document]
Comment (tolgaarikan) [reply 8-ui-tests.md]: Makes sense 😄

File: 8-ui-tests.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    func testSignup() {
+        // * We launch to app to start testing. This is the most time consuming step
+        // so we will combine all 3 tests and do them in one launch.
+        app.launch()
+      
+        // * Preparing the views that we will check later if they exist
+        let landingView = app.otherElements["LandingView"]
+        let signupView = app.otherElements["SignupCardView"]
+        let signupLoadingView = app.otherElements["SignupLoadingView"]
```
Comment (tungfam) [inline 8-ui-tests.md]: do i understand it right that string in `""` is `accessibilityIdentifier`? if yes, maybe let's say about it shortly to make it clearer.

File: 8-ui-tests.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        // * We use waitForExistence method to see if landingView is visible after
+        // dismissing SignupCardView
+
+        // Test Closing with Swipe * Second Test
+        app.openSignupScreen() // * Again we navigate to the screen we want to test
+        
+        XCTAssert(signupView.waitForExistence(timeout: 5))
+        // * Again we use .waitForExistence method to see if it appeared.
+
+        signup.swipeDown()
```
Comment (tungfam) [inline 8-ui-tests.md]: what's the instance `signup`?

File: 8-ui-tests.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+But this is only taking into consediration the tools that Xcode gives us. We might be able to find an alternative(there are already some that seem promising like iOS Snapshot Test Case and EarlGrey) that is easier to use and let's us do faster implementation with less time wasted on querying the UI elements and on other XCUITest framework weirdness. And at that point it might be plausible to give this responsibility to the QA team.
```
Comment (tungfam) [inline 8-ui-tests.md]: `consideration` typo)

**Pull request merged**